### ISSUE TITLE
A Changelog would be useful

### ISSUE THREAD

**Issue by jules2689** [octobox/octobox#584]
Are you experiencing this issue in octobox.io or your own instance?
 - [ ] octobox.io
 - [x] My instance

🐞 **Problem**

Right now when pulling in upstream to update my instance, I have to go and parse through commits to find relevant and actionable items.

Recently this was exasperated by many emoji being added (clogging the diff) and sidekiq (something I need to action).

🎯 **Goal**

Make upgrading instances easier.

💡 **Possible solutions**

Use probot or similar to automate a changelog?
- https://github.com/uber-workflow/probot-app-release-notes
- https://github.com/mikz/probot-changelog
- https://github.com/semantic-release/semantic-release

Or suggest that all PRs decide to file their change under `Major` `Minor` etc

I know that this would be easier with proper release cycles, which we don't do now - but maybe there's a happy medium?

cc @chrisarcand @andrew @tarebyte

**Comment by chrisarcand** [comment]
Hehe, I was *literally* just thinking this same thought when I was looking at what's been merged recently and thinking 'I should really update my box before I forget all the things I need to do for it...' 👍

**Comment by andrew** [comment]
At the moment, every commit merged into master generates a new docker image. 

Feels like having a more curated, hand written changelog for significant changes would be the way to go?

**Comment by andrew** [comment]
I'm also not a big fan of encoding the type of change into the commit message, ala semantic-release, mostly because it becomes a huge burden on new contributors, asking them to rebase their commits to change the messages etc

**Comment by jules2689** [comment]
Yea, that makes sense. The big thing that is a "blocker" here is that there are no "releases" as it's continuously deployed (more or less). There's much more succinct and obvious changelogs for an app that has a release.

**Comment by andrew** [comment]
I guess we could start tagging known-good releases on master on a regular basis and encouraging people to pull from tagged releases rather than just master?

**Comment by chrisarcand** [comment]
I agree on the cons against autogenerated changelogs, or at least the ones that claim no human interaction is necessary; they just create pain in my experience. 

Yeah tagging sounds fine to me. Semver might be hard to follow for an application; the 'breaking changes' bit gets really fuzzy and I don't want to get in to lengthy, pedantic discussions about what constitutes 'breaking'. If we start tagging it should be expected that it's going to look a lot like Firefox versioning, where the major number is incremented very often and is really the release number. (35.0, 35.0.1, 35.0.2, 36.0....) Thoughts?

**Comment by andrew** [comment]
Yeah I have no problem with just rolling release numbers, semver doesn't make much sense when we don't have many people interfacing with it directly.

**Comment by jules2689** [comment]
What is we made release logs on a weekly or bimonthly basis?

**Comment by jules2689** [comment]
How about this:
- We make a manual "first release" now
- We set up a bot that runs once every two weeks that auto tags the current commit
- We make an issue to fill out the tagged release with a "changelog"

This way, it's automating the process to actually force it to happen and manually doing the important parts.

**Comment by andrew** [comment]
Sounds like a plan!

**Comment by andrew** [comment]
Once we have something like that, would be great to have a way of browsing the changelog from the web ui, maybe even suggesting updating for people running their own instances?

**Comment by andrew** [comment]
We should also use the https://keepachangelog.com standard format

**Comment by jules2689** [comment]
Here's our baseline then: https://github.com/octobox/octobox/releases/tag/e62e539

**Comment by chrisarcand** [comment]
Nice, I dig this a lot.

**Comment by andrew** [comment]
:+1:

**Comment by andrew** [comment]
Closing this as we have a changelog now
